patches/tModLoader/Terraria/NPC.cs.patch

```diff
@@ -562,10 +575,11 @@
  					Main.npc[i].active = false;
  					NetMessage.SendData(23, -1, -1, null, i);
  				}
-@@ -59769,6 +_,7 @@
+@@ -59769,6 +_,8 @@
  			if (type == 551)
  				itemType = 3860;
  
++			//Left in for compatibility. It is now in ItemDropDatabase.RegisterBosses
 +			NPCLoader.BossBag(this, ref itemType);
```

Comment:
Is this advisable? Is this hook actually usable for anything here?

Reply:
The method actually doesn't contain any of the new 1.4 bosses, for the record. And yeah I just left it in cause I didn't know if I should remove it or not, don't know what tmls policies on keeping legacy tml methods inside legacy vanilla methods is.

Reply:
Actually I think it can be removed, given it is auto-registered AND auto-dropped through the drop handler. Mods used to have to call npc.DropBossBags() in NPCLoot manually, if they would still do it in 1.4 in the OnKill hook, bags would drop twice.

Reply:
Upon further checking vanilla code regarding bags, I noticed one boss not utilizing the logic this PR would use: Eater of Worlds. Its in practically every respect unique, there's exceptions for it everywhere, no surprise this is also the case here. If a mod were to implement a boss with the same logic (any segment could potentially drop the bag, but only one bag should be dropped), it's easy to remove the drop rule from the NPC and insert a custom boss bag rule (the same one EoW uses).